src/main/java/roomescape/controller/LoginController.java

```diff
@@ -0,0 +1,117 @@
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class LoginController {
+
+	private static final int COOKIE_MAX_AGE = 3600;
+
+	private static final String COOKE_PATH = "/";
+
+	private static final String TOKEN = "token";
+
+	private static final String ROLE = "role";
+
+	private final MemberService memberService;
+
+	private final JwtTokenProvider jwtTokenProvider;
+
+	LoginController(MemberService memberService, JwtTokenProvider jwtTokenProvider) {
+		this.memberService = memberService;
+		this.jwtTokenProvider = jwtTokenProvider;
+	}
+
+	@PostMapping("/login")
+	public ResponseEntity<Void> login(@RequestBody LoginRequest request, HttpServletResponse response) {
+		LoginRequest.validateLoginInfo(request);
+		try {
+			// TODO: STEP2 HandlerMethodArgumentResolver 를 통해서 리팩터링 예정
+			LoginRequest loginRequest = new LoginRequest(request.email(), PasswordEncoder.encode(request.password()));
+			MemberResponse memberResponse = this.memberService.findMemberByLoginRequest(loginRequest);
+
+			String token = this.jwtTokenProvider.createToken(memberResponse);
+			var cookie = this.createCookie(token, COOKIE_MAX_AGE);
+			response.addCookie(cookie);
+
+			return ResponseEntity.ok().build();
+		}
+		catch (RoomEscapeException ex) {
+			if (ex.getErrorCode().equals(ErrorCode.NOT_FOUND_MEMBER)) {
+				return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+			}
+			else {
+				return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+			}
+		}
+
+	}
+
+	@GetMapping("/login/check")
+	public ResponseEntity<LoginCheckResponse> check(HttpServletRequest request) {
+		return ResponseEntity.ok().body(this.findRoleByToken(request.getCookies()));
+	}
+
+	@PostMapping("/logout")
+	public ResponseEntity<Void> logout(HttpServletRequest request, HttpServletResponse response) {
+		var tokenCookie = this.extractTokenFromCookie(request.getCookies());
+		Cookie cookie = this.createCookie(tokenCookie, 0);
+		response.addCookie(cookie);
+		return ResponseEntity.ok().build();
+	}
+
+	// TODO: STEP2 에서 리팩터링 예정
+	public Cookie createCookie(String token, int maxAge) {
+		Cookie cookie = new Cookie(TOKEN, token);
+		cookie.setHttpOnly(true);
+		cookie.setPath(COOKE_PATH);
+		cookie.setMaxAge(maxAge);
+		return cookie;
+	}
+
+	private LoginCheckResponse findRoleByToken(Cookie[] cookies) {
+
+		if (cookies == null || cookies.length == 0) {
+			throw new RoomEscapeException(ErrorCode.NEEDS_LOGIN);
+		}
+		var token = extractTokenFromCookie(cookies);
+
+		Claims role = this.jwtTokenProvider.validateToken(token);
+		String roleName = role.get(ROLE, String.class);
+
+		return new LoginCheckResponse(roleName);
+	}
+
+	private String extractTokenFromCookie(Cookie[] cookies) {
+		if (cookies == null || cookies.length == 0) {
+			return "";
+		}
+
+		return Arrays.stream(cookies)
+			.filter((cookie) -> TOKEN.equals(cookie.getName()))
```

Comment:
```suggestion
			.filter(cookie -> TOKEN.equals(cookie.getName()))
```

람다 형식을 사용할 때 파라미터가 하나인 경우, 이렇게 괄호 없이 쓸 수도 있어요.
람다가 사용된 다른 곳들도 한 번 봐주세요!

Reply:
아 이 부분은 spring check style 을 적용해서 그런데요.

매개 변수가 하나인 경우에도 괄호를 사용하면 코드의 가독성이 향상되기도하고,
다중 매개 변수를 사용할 때도 일관된 스타일을 유지시킬 수 있어요!

그래서 spring check style 에서는 매개 변수가 하나 일 때 괄호로 감싸지 않으면 경고를 띄워주고,
일관성 있는 스타일 유지를 위해서 적용했어요~

![image](https://github.com/next-step/spring-roomescape-auth/assets/37852701/5a0e679f-782c-49d8-a812-80383d860ce3)

Reply:
그렇군요! 좋습니다 👍 👍 👍